addons/hw_l10n_eg_eta/controllers/main.py

```diff
@@ -0,0 +1,144 @@
+                cert = next(session.get_objects({Attribute.CLASS: ObjectClass.CERTIFICATE}))
+                payload = {
+                    'otp': otp,
+                    'user': user,
+                    'drive': drive,
+                    'certificate': base64.b64encode(cert[Attribute.VALUE]).decode('utf-8')
+                }
+                try:
+                    requests.post(url + '/l10n_eg_edi_eta/certificate', data=payload, timeout=10)
+                except requests.exceptions.Timeout:
+                    return self.timeout_error()
+                except requests.exceptions.ConnectionError:
+                    return self.connection_error()
+
+                return json.dumps({'status': 'success'})
+        except PinIncorrect:
+            return self.get_error_template('Incorrect pin', 'الرمز غير صحيح')
+        except Exception as ex:
+            return self.get_error_template(str(ex), '')
+
+    @http.route('/hw_l10n_eg_eta/sign', type='http', auth='none', cors='*', csrf=False, save_session=False)
+    def eta_sign(self, url, invoices, pin, drive, user, otp):
+        try:
+            invoice_dict = dict()
+
+            payload = {
+                'token': otp,
+                'user': user,
+                'drive': drive,
+                'invoice_ids': invoices
+            }
+
+            try:
+                resp = requests.get(url + '/l10n_eg_edi_eta/get_invoice', params=payload, timeout=10)
+            except requests.exceptions.Timeout:
+                return self.timeout_error()
+            except requests.exceptions.ConnectionError:
+                return self.connection_error()
+
+            if resp.status_code != 200:
+                return self.get_error_template('Internal server error', 'خطأ في الخادم الداخلي')
+
+            resp = resp.json()
+
+            if resp.get('errors'):
+                return self.get_error_template(resp.get('errors'), '')
+
+            token, error = self._get_token()
+            if error:
+                return error
+
+            with token.open(user_pin=pin) as session:
+                cert = next(session.get_objects({Attribute.CLASS: ObjectClass.CERTIFICATE}))
+                priv_key = next(session.get_objects({Attribute.CLASS: ObjectClass.PRIVATE_KEY, Attribute.ID: cert[Attribute.ID]}))
+
+                for invoice, eta_inv in resp.items():
+                    to_sign = base64.b64decode(eta_inv)
+                    signed_data = priv_key.sign(to_sign, mechanism=pkcs11.mechanisms.Mechanism.SHA256_RSA_PKCS)
+                    invoice_dict[invoice] = base64.b64encode(signed_data).decode('utf-8')
+
+            payload = {
+                'token': otp,
+                'user': user,
+                'drive': drive,
+                'invoices': json.dumps(invoice_dict),
+            }
+            try:
+                resp = requests.post(url + '/l10n_eg_edi_eta/submit_signed_invoices', data=payload, timeout=10)
```

Comment:
Same here: it would be weird that this line gives an PinIncorrect, so it would be better to make the try..except/with/... a flatter structure.